deploy/charts/csi-driver/templates/clusterrole.yaml

```diff
@@ -8,3 +8,9 @@ rules:
 - apiGroups: ["cert-manager.io"]
   resources: ["certificaterequests"]
   verbs: ["watch", "create", "delete", "list"]
+{{- if and (.Capabilities.APIVersions.Has "security.openshift.io/v1") (.Values.openshift.securityContextConstraint.enabled) }}
+- apiGroups: ["security.openshift.io"]
+  resources: ["securitycontextconstraints"]
+  resourceNames: [{{ .Values.openshift.securityContextConstraint.name | quote }}]
+  verbs: ["use"]
+{{- end }}
```

Comment:
question: I have a vague recollection that the Helm `.Capabilities` stuff doesn't play well with `helm template` which I think we generally want to support.

Running locally, I can see that `helm template` on this PR will not add the openshift RBAC, which I guess is expected although if I were trying to render the YAML to apply to an OpenShift cluster I'd find that annoying. I think it's possible to tell `helm template` what capabilities are available but I think it's a pain IIRC

Would it be simpler all-round to have `openshift.securityContextConstraint.enabled` default to false and to document that OpenShift users should enable it? It'd then be very clear how this would work with `helm template`.

For bonus points we could emit a warning if `.Capabilities.APIVersions.Has "security.openshift.io/v1"` is true but `.Values.openshift.securityContextConstraint.enabled` is false?

Also